tools/data/textrecog/funsd_converter.py

```diff
@@ -0,0 +1,203 @@
+
+    return files
+
+
+def collect_annotations(files, nproc=1):
+    """Collect the annotation information.
+
+    Args:
+        files(list): The list of tuples (image_file, groundtruth_file)
+        nproc(int): The number of process to collect annotations
+
+    Returns:
+        images(list): The list of image information dicts
+    """
+    assert isinstance(files, list)
+    assert isinstance(nproc, int)
+
+    if nproc > 1:
+        images = mmcv.track_parallel_progress(
+            load_img_info, files, nproc=nproc)
+    else:
+        images = mmcv.track_progress(load_img_info, files)
+
+    return images
+
+
+def load_img_info(files):
+    """Load the information of one image.
+
+    Args:
+        files(tuple): The tuple of (img_file, groundtruth_file)
+
+    Returns:
+        img_info(dict): The dict of the img and annotation information
+    """
+    assert isinstance(files, tuple)
+
+    img_file, gt_file = files
+    assert osp.basename(gt_file).split('.')[0] == osp.basename(img_file).split(
+        '.')[0]
+    # read imgs while ignoring orientations
+    img = mmcv.imread(img_file, 'unchanged')
+
+    img_info = dict(
+        file_name=osp.join(osp.basename(img_file)),
+        height=img.shape[0],
+        width=img.shape[1],
+        segm_file=osp.join(osp.basename(gt_file)))
+
+    if osp.splitext(gt_file)[1] == '.json':
+        img_info = load_json_info(gt_file, img_info)
+    else:
+        raise NotImplementedError
+
+    return img_info
+
+
+def load_json_info(gt_file, img_info):
+    """Collect the annotation information.
+
+    Args:
+        gt_file(str): The path to ground-truth
+        img_info(dict): The dict of the img and annotation information
+
+    Returns:
+        img_info(dict): The dict of the img and annotation information
+    """
+
+    annotation = mmcv.load(gt_file)
+    anno_info = []
+    for form in annotation['form']:
+        for ann in form['words']:
+
+            # Ignore illegible samples
+            if len(ann['text']) == 0:
+                continue
+
+            x1, y1, x2, y2 = ann['box']
+            x = max(0, min(math.floor(x1), math.floor(x2)))
+            y = max(0, min(math.floor(y1), math.floor(y2)))
+            w, h = math.ceil(abs(x2 - x1)), math.ceil(abs(y2 - y1))
+            bbox = [x, y, x + w, y, x + w, y + h, x, y + h]
+            word = ann['text']
+
+            anno = dict(bbox=bbox, word=word)
+            anno_info.append(anno)
+
+    img_info.update(anno_info=anno_info)
+
+    return img_info
+
+
+def generate_ann(root_path, split, image_infos, preserve_vertical):
+    """Generate cropped annotations and label txt file.
+
+    Args:
+        root_path(str): The root path of the dataset
+        split(str): The split of dataset. Namely: training or test
+        image_infos(list[dict]): A list of dicts of the img and
+        annotation information
+        preserve_vertical(bool): Whether to preserve vertical texts
```

Comment:
```suggestion
        root_path (str): The root path of the dataset
        split (str): The split of dataset. Namely: training or test
        image_infos (list[dict]): A list of dicts of the img and
            annotation information
        preserve_vertical (bool): Whether to preserve vertical texts
```
Please read through the docstring section in the knowledge base for the style guide

Reply:
Ok, I will read and correct the style issues.